retis/src/cli/cli.rs

```diff
@@ -174,33 +163,34 @@ pub(crate) struct MainConfig {
         help = "Comma separated list of profile names to apply"
     )]
     pub(crate) profile: Vec<String>,
-    #[arg(
-        long,
-        help = "Path to kernel configuration (e.g. /boot/config-6.3.8-200.fc38.x86_64; default: auto-detect)"
-    )]
-    pub(crate) kconf: Option<PathBuf>,
 }
 
-/// ThinCli handles the first (a.k.a "thin") round of Command Line Interface parsing.
-///
-/// During this phase, SubCommands can be added. After all SubCommands have been added, the build()
-/// method will run the thin CLI parsing that does not perform dynamic subcommand argument
-/// validation and yield a FullCli object to represent the results.
-#[derive(Debug)]
-pub(crate) struct ThinCli {
+#[derive(Debug, Default)]
+pub(crate) struct RetisCli {
     subcommands: Vec<Box<dyn SubCommand>>,
 }
 
-impl ThinCli {
-    /// Allocate and return a new ThinCli object that will parse the command arguments.
+impl RetisCli {
+    /// Allocate and return a new RetisCli object that will parse the command arguments.
     pub(crate) fn new() -> Result<Self> {
-        Ok(ThinCli {
-            subcommands: Vec::new(),
-        })
+        let mut cli = RetisCli::default();
+        cli.add_subcommand(Box::new(Collect::new()?))?;
+        cli.add_subcommand(Box::new(Print::new()?))?;
+        cli.add_subcommand(Box::new(Sort::new()?))?;
+        #[cfg(feature = "python")]
+        cli.add_subcommand(Box::new(PythonCli::new()?))?;
+        cli.add_subcommand(Box::new(Pcap::new()?))?;
+        cli.add_subcommand(Box::new(Inspect::new()?))?;
+        cli.add_subcommand(Box::new(ProfileCmd::new()?))?;
+        cli.add_subcommand(Box::new(Complete::new()?))?;
+
+        #[cfg(feature = "benchmark")]
+        cli.add_subcommand(Box::new(Benchmark::new()?))?;
+
+        Ok(cli)
```

Comment:
This means we don't have logging here for all the sub-commands initialization. I don't think that's necessarily an issue, logging usually happens later as initialization either work or fail (not that there couldn't be some use cases).

If we really want to have logging before maybe sub-commands can be added after an initial parsing of the main args? Otherwise maybe we could add a comment in this function so future additions are aware of this limitation?

Reply:
Yeah, we don't. We didn't have log access before so I don't think anything changed here. But sure, I'll add a comment for future reference.